docs/modules/ROOT/pages/explanations/decisions/capacity-alerting.adoc

```diff
@@ -0,0 +1,111 @@
+= Handling of Capacity Alerts for Incident Prevention
+
+== Problem
+
+All Databases by VSHN have limited capacity such as storage and memory capacity, or a limit on concurrent connections.
+Depending on the database type running out of capacity can lead to downtime or even to data loss.
+We need a way to predict capacity issues, so we can react and prevent an incident.
+
+Writing these predictive capacity alerts is generally a solved problem.
+We can write Prometheus queries that predict the resource usage and alert based on the predicted usage.
+The main question is how we want to handle these alerts.
+
+From experience from similar systems, we know that in nearly all cases capacity issues can only be resolved by the service users themselves, either by freeing up resources or by scaling up the instance.
+So first and foremost we need a process to inform the service user about potential capacity issues.
+
+== Proposals
+
+=== Alert Responsible Ops
+
+The simplest approach would be to alert the engineer that does https://handbook.vshn.ch/role_responsibleops.html[responsible ops^].
+This is how we are used to handling alerts and how we handled this in previous systems.
+
+In practice this means we will deploy fixed Prometheus alert rules and Alertmanager configs in every database namespace that will send alerts to Opsgenie.
+
+The only open question is how we securely distribute the necessary Opsgenie secrets to the database namespace.
+This can be solved by https://github.com/vshn/espejo[espejo] or through a custom controller, but we need to be aware that increase the potential attack surface by distributing Opsgenie credentials to many namespaces.
+
+=== Expose User-Workload Monitoring
+
+APPUiO Cloud and APPUiO Managed OpenShift come with a versatile user-workload monitoring system that allows a user to specify their own Alertmanager configuration on a namespace level.
+
+We could expose the Alertmanager configuration of this system directly in the database claim and the composition will then create an `AlertManagertConfig` resource in the database namespace.
+This would enable a service user to decide for themselves how they want to be alerted about capacity issues.
+
+With such a system the database claim could look similar to the following, which would configure all alerts to be sent to the service users slack instance.
```

Comment:
I like this as a first approach, but I feel like this will lead to a lot of duplicated configurations for the customer. We should look into the possibility of having one or multiple such alertmanager configs per user namespace and then distribute them to the instance namespaces.

That should not be too complicated and makes the UX for the user much better IMHO.

Reply:
We decided in the team that this might be a good idea idea, but we'll start with the proposed, simpler idea for now.